### DISCUSSION TITLE
Query().destroy not supporting timeout although it's a bulk operation

### DISCUSSION THREAD

**Discussion by toddhalfpenny** [jsforce/jsforce#1438] [Ideas]
Seeing as the `query().destroy()` in turn is a bulk operation it would be good to add a timeout to this.

I'd be happy to work on this and create a PR if some guidance could be given.

**Comment by cristiand391** [comment]
when doing a bulk update/query like this:
https://jsforce.github.io/document/#update-delete-queried-records

it uses the bulk APIs so you can just set it like when using the `bulk` methods:
https://jsforce.github.io/document/#bulk-api

> NOTE: In some cases for large data sets, a polling timeout can occur. When loading large data sets, consider changing Bulk.pollTimeout and Bulk.pollInterval property value, or using the one of the calls above with the built in batch.poll() or polling manually.
conn.bulk.pollTimeout = 25000; // Bulk timeout can be specified globally on the conn

and `conn.bulk2.pollTimeout` if using the Bulk V2 API.

That said, we could expose the polling options as args to `destroy/update` here:
https://jsforce.github.io/jsforce/classes/query.Query.html#destroy
https://jsforce.github.io/jsforce/types/query.QueryDestroyOptions.html

to avoid setting  "global" polling options that affect other bulk operations.
